### PR TITLE
Create checkDuplicateCommands.ts

### PR DESCRIPTION
**Pull request by EndlessM0DZ**
This will warn users of duplicate command names and give the respective locations of the files. I tested this code in a simple test environment in JS, so it may need more testing and tweaks.

### CODE REVIEW

File: packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+       let duplicatePairs = tempCommands.filter((obj, index) => {
+        if(obj.data.name === tempCommands[i].data.name){
+          indexes.push(index)
+          return true
+        }
+        else return false
+       })
+       if(duplicatePairs.length > 1) duplicates.push(duplicatePairs)
+       else nonduplicates.push(duplicatePairs)
+       for(let j = indexes.length - 1; j > -1; j--){
```
Comment (twlite) [inline packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts:16]: can we simplify this more?

File: packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        duplicates,
+        nonduplicates
+    }
+  }
+export function warnDuplicatesIfAny(commands: CommandFileObject[]){
+  let { duplicates, nonduplicates } = checkDuplicateCommands(commands)
+  for(let i = 0; i < duplicates.length; i++){
+    let myString = `You have duplicate command names with command: ${colors.red(duplicates[i][0].data.name)} in the following files:`
+    for(let j = 0; j < duplicates[i].length; j++){
+        myString += `\n${duplicates[i][j].filePath}`
```
Comment (twlite) [inline packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts:30]: why do we need nested loops? Can't we just use `Array.prototype.join`?

**Review by twlite — CHANGES_REQUESTED** [review]

File: packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts (hunk 2 above)
Comment (EndlessM0DZ) [reply packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts]: Yes, Array.prototype.join is a better approach, I just modified that.

File: packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts (hunk 1 above)
Comment (EndlessM0DZ) [reply packages/commandkit/src/handlers/command-handler/functions/checkDuplicateCommands.ts]: The logic now creates an obj that will have a key and hold the CommandFileObjects in an array then check if the length of that key is above one. This code does look a lot cleaner now.

**Review by twlite — COMMENTED** [review]
this code does not have any effect on the framework. You added the functions but it has not been used at all.

**Pull request closed without merging**